ee/loader/src/main.c

```diff
@@ -1214,6 +1221,8 @@ int main(int argc, char *argv[])
         // Load fragfile
         if (fhi_bdm_add_file(set_fhi_bdm, FHI_FID_ATA0, sATA0File) < 0)
             return -1;
+        // if ata 0 is ok, load HDD ID file
+        fhi_bdm_add_file(set_fhi_bdm, FHI_FID_ATAID0, sATAID0File);
```

Comment:
Add `if (sATA0IDFile != NULL)` to prevent this from loading when we do not want to, and also check for the return value and `return -1` when the file must be loaded (but fails to load).

Reply:
Done